chore: add v3 instructions (#72)
traefik/traefik-library-image: README.md

![Traefik Logo](https://docs.traefik.io/assets/img/traefik.logo.png)

[Traefik](https://traefik.io) is a modern HTTP reverse proxy and load balancer that makes deploying microservices easy.

Traefik integrates with your existing infrastructure components ([Docker](https://www.docker.com/), [Swarm mode](https://docs.docker.com/engine/swarm/), [Kubernetes](https://kubernetes.io), [Marathon](https://mesosphere.github.io/marathon/), [Consul](https://www.consul.io/), [Etcd](https://coreos.com/etcd/), [Rancher](https://rancher.com), [Amazon ECS](https://aws.amazon.com/ecs), ...) and configures itself automatically and dynamically.

Pointing Traefik at your orchestrator should be the *only* configuration step you need.

## Traefik v3 - Example usage

Enable `docker` provider and dashboard UI:

```yml
## traefik.yml

# Docker configuration backend
providers:
  docker:
    defaultRule: "Host(`{{ trimPrefix `/` .Name }}.docker.localhost`)"

# API and dashboard configuration
api:
  insecure: true
```

Start Traefik v3:

```sh
docker run -d -p 8080:8080 -p 80:80 \
  -v $PWD/traefik.yml:/etc/traefik/traefik.yml \

>   -v /var/run/docker.sock:/var/run/docker.sock \

Reviewer: ```suggestion
  -v /var/run/docker.sock:/var/run/docker.sock \
```
I'm not sure to understand why it should be mounted
Is this still recommended with v3 ?

Author: I can run the image successfully without that command @kevinpollet is this still recommended?

>   traefik:v3.2

Reviewer: ```suggestion
  traefik:v3
```

Reviewer 2: Why not using the v3 tag here?
I think for the latest major this makes sense.

![Traefik Logo](https://docs.traefik.io/assets/img/traefik.logo.png)

> [Traefik](https://traefik.io) is a modern HTTP reverse proxy and load balancer that makes deploying microservices easy.

Reviewer 3: ```suggestion
[Traefik](https://traefik.io) is a modern HTTP reverse proxy and ingress controller that makes deploying microservices easy.
```

> Traefik integrates with your existing infrastructure components ([Docker](https://www.docker.com/), [Swarm mode](https://docs.docker.com/engine/swarm/), [Kubernetes](https://kubernetes.io), [Marathon](https://mesosphere.github.io/marathon/), [Consul](https://www.consul.io/), [Etcd](https://coreos.com/etcd/), [Rancher](https://rancher.com), [Amazon ECS](https://aws.amazon.com/ecs), ...) and configures itself automatically and dynamically.

Reviewer 3: ```suggestion
Traefik integrates with your existing infrastructure components ([Kubernetes](https://kubernetes.io), [Docker](https://www.docker.com/), [Swarm](https://docs.docker.com/engine/swarm/), [Consul](https://www.consul.io/), [Nomad](https://www.nomadproject.io/), [Etcd](https://coreos.com/etcd/), [Rancher](https://rancher.com), [Amazon ECS](https://aws.amazon.com/ecs), ...) and configures itself automatically and dynamically.
```

![Traefik Logo](https://docs.traefik.io/assets/img/traefik.logo.png)

[Traefik](https://traefik.io) is a modern HTTP reverse proxy and ingress controller that makes deploying microservices easy.

> Traefik integrates with your existing infrastructure components ([Kubernetes](https://kubernetes.io), [Docker](https://www.docker.com/), [Swarm](https://docs.docker.com/engine/swarm/), [Consul](https://www.consul.io/), [Nomad](https://www.nomadproject.io/), [Etcd](https://coreos.com/etcd/), [Rancher](https://rancher.com), [Amazon ECS](https://aws.amazon.com/ecs), ...) and configures itself automatically and dynamically.

Reviewer 4: ```suggestion
Traefik integrates with your existing infrastructure components ([Kubernetes](https://kubernetes.io), [Docker](https://www.docker.com/), [Swarm](https://docs.docker.com/engine/swarm/), [Consul](https://www.consul.io/), [Nomad](https://www.nomadproject.io/), [Etcd](https://coreos.com/etcd/), [Amazon ECS](https://aws.amazon.com/ecs), ...) and configures itself automatically and dynamically.
```

Author: this deletes Rancher too, is rancher not supported also?

Reviewer 4: Indeed, Rancher support was removed in v3: https://doc.traefik.io/traefik/migration/v2-to-v3-details/#rancher-v1-provider.
Because we already removed Marathon (for the same reason), I think we should also remove Rancher from this mention.

Author: Okay, got it

  insecure: true
```

Start Traefik v3:

```sh
docker run -d -p 8080:8080 -p 80:80 \
  -v $PWD/traefik.yml:/etc/traefik/traefik.yml \
  -v /var/run/docker.sock:/var/run/docker.sock \
  traefik:v3
```  

Start a backend server using the `traefik/whoami` image:

```sh
docker run -d --name test traefik/whoami
```

> Access the whoami Service through Traefik via the defined rule `test.docker.localhost`:

Reviewer 2: ```suggestion
Access the whoami service through Traefik via the defined rule `test.docker.localhost`:
```

```console
$ curl test.docker.localhost
Hostname: 0693100b16de
IP: 127.0.0.1
IP: ::1
IP: 192.168.215.4
RemoteAddr: 192.168.215.3:57618
GET / HTTP/1.1
Host: test.docker.localhost
User-Agent: curl/8.7.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 192.168.215.1
X-Forwarded-Host: test.docker.localhost
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: 8a37fd4f35fb
X-Real-Ip: 192.168.215.1
```

Access the Traefik Dashboard:

Open your web browser and navigate to `http://localhost:8080` to access the Traefik dashboard.
This will provide an overview of routers, services, and middlewares.

![Dashboard UI](https://raw.githubusercontent.com/traefik/traefik/v3.2/docs/content/assets/img/webui-dashboard.png)

## Traefik v2 - Example usage

Enable `docker` provider and dashboard UI:

```yml
## traefik.yml

# Docker configuration backend
providers:
  docker:

>     defaultRule: "Host(`{{ trimPrefix `/` .Name }}.docker.localhost`)"

Reviewer 2: @sheddy-traefik @emilevauge One question, we are promoting the default rule configuration here to have the router created automatically/magically for the whoami backend (an option which IMHO is not widely used). I wonder if we should change this and add the router labels to the whoami container deployment? wdyt?

Author: i don't think it matters whether it is widely used or not, what matters is that is it recommended? is it possible? are there any drawbacks to using this over router labels. Also we have to keep in mind that this is a demo and not something people look at to model their production environments after. 

Let me know what you think

Reviewer 2: This is working and there are no drawbacks (just lack of controls), you are right, for a demo I think this is ok.

# API and dashboard configuration
api:
  insecure: true
```

Start Traefik v2:

```sh
docker run -d -p 8080:8080 -p 80:80 \
-v $PWD/traefik.yml:/etc/traefik/traefik.yml \
-v /var/run/docker.sock:/var/run/docker.sock \
traefik:v2.11
```

Start a backend server using the `traefik/whoami` image:

```sh
docker run -d --name test traefik/whoami
```

> Access the whoami Service through Traefik via the defined rule `test.docker.localhost`:

Reviewer 2: ```suggestion
Access the whoami service through Traefik via the defined rule `test.docker.localhost`:
```

  insecure: true
```

Start Traefik v3:

```sh
docker run -d -p 8080:8080 -p 80:80 \
  -v $PWD/traefik.yml:/etc/traefik/traefik.yml \
  -v /var/run/docker.sock:/var/run/docker.sock \
  traefik:v3.2
```  

Start a backend server using the `traefik/whoami` image:

```sh
docker run -d --name test traefik/whoami
```

Access the whoami Service through Traefik:

```console
$ curl test.docker.localhost
Hostname: 0693100b16de
IP: 127.0.0.1
IP: ::1
IP: 192.168.215.4
RemoteAddr: 192.168.215.3:57618
GET / HTTP/1.1
Host: test.docker.localhost
User-Agent: curl/8.7.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 192.168.215.1
X-Forwarded-Host: test.docker.localhost
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: 8a37fd4f35fb
X-Real-Ip: 192.168.215.1
```

Access the Traefik Dashboard:

Open your web browser and navigate to `http://localhost:8080` to access the Traefik dashboard.
This will provide an overview of routers, services, and middlewares.

> ![Web UI](https://raw.githubusercontent.com/traefik/traefik/v3.2/docs/content/assets/img/webui-dashboard.png)

Reviewer 2: ```suggestion
![Dashboard UI](https://raw.githubusercontent.com/traefik/traefik/v3.2/docs/content/assets/img/webui-dashboard.png)
```

> ## Traefik v2 - Example usage

Reviewer 2: The wording and formatting between v2 and v3 are not identical.
Does it make sense to align them? (the new wording in v3 sounds good, so I would do the same for the v2 section)

Author: I think it does make sense, for consistency, I agree with you.

Author: Updated the PR @kevinpollet

Enable `docker` provider and web UI:

```yml
## traefik.yml

# Docker configuration backend
providers:
  docker:
    defaultRule: "Host(`{{ trimPrefix `/` .Name }}.docker.localhost`)"

# API and dashboard configuration
api:
  insecure: true
```

Start Traefik v2:

> ```bash

Reviewer 2: ```suggestion
```sh
```

docker run -d --name test traefik/whoami
```

And finally, you can access to your `whoami` server throught Traefik, on the domain name `test.docker.localhost`:

```console

> # $ curl --header 'Host:test.docker.localhost' 'http://localhost:80/'

Reviewer 2: Not sure that this line is necessary.

> # $ curl --header 'Host:test.docker.localhost' 'http://localhost:80/'

Reviewer 2: ```suggestion
```

Not sure that this line is necessary.

  domain = "docker.localhost"
```

> Start Traefik v1:

Reviewer 3: Section to be removed
